util/tracer_nvbit/tracer_tool/traces-processing/post-traces-processing.cpp

```diff
@@ -109,46 +113,76 @@ int main(int argc, char **argv) {
   ifstream ifs;
   ofstream ofs;
 
-  ifs.open(kernellist_filepath.c_str());
-  ofs.open((string(kernellist_filepath) + ".g").c_str());
-
-  if (!ifs.is_open()) {
-    cerr << "Unable to open file: " << kernellist_filepath << endl;
+  // We can now pass a directory or a file as the input argument 
+  std::filesystem::path p(filepath);
+  if (std::filesystem::is_directory(p))
+  {
+    for (const auto & entry : std::filesystem::directory_iterator(p))
+    {
+      std::string filename = entry.path().filename();
+      if (filename.find("kernelslist") != std::string::npos)
+      {
+        kernelslist_list.push_back(entry.path().string());
+      }
+    }
+  }
+  else if (std::filesystem::is_regular_file(p))
+  {
+    kernelslist_list.push_back(kernellist_filepath);
+  }
+  else
+  {
+    cerr << "Invalid file path\n";
     return 1;
   }
 
-  string directory(kernellist_filepath);
-  const size_t last_slash_idx = directory.rfind('/');
-  if (std::string::npos != last_slash_idx) {
-    directory = directory.substr(0, last_slash_idx);
-  }
+  for (auto kernellist_filepath : kernelslist_list){
+    string directory(kernellist_filepath);
+    const size_t last_slash_idx = directory.rfind('/');

```

Comment:
So fancy, but this is still the best way to find the last dir :)
I always through Mahmoud should haver written this in python, but I think he just didn't want to learn